### PR TITLE
Add actions to Music Assistant

### PR DESCRIPTION
**Pull request by OzGav**
## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [X] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/133521

## Checklist

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
	- Introduced `music_assistant.search` action for global media searches with various customizable attributes.
	- Added `music_assistant.get_library` action for local library searches with enhanced filtering options.
	- Implemented `music_assistant.get_queue` action for retrieving the current media queue.

### CODE REVIEW

File: source/_integrations/music_assistant.markdown [hunk 1]
```diff
@@ -135,6 +135,89 @@ Transfer the player's queue to another player.
...
+  - **Optional**: No.
+  - **Description**: The Music Assistant instance that the search will be performed upon. Allows for multiple servers to be running.
+  - **Example**: `Music Assistant`
+- **Data attribute**: `name`
+  - **Optional**: No.
+  - **Description**: The name/title to search for.
+  - **Example**: `We Are The Champions`
+- **Data attribute**: `media_type`
+  - **Optional**: Yes.
+  - **Description**: The type(s) of content to search for. Select from artist, album, track, radio, or playlist. All types if omitted.
```
Comment (marcelveldt) [inline source/_integrations/music_assistant.markdown:152]:
```suggestion
  - **Description**: The type(s) of content to search for. All types if omitted.
```
The available options are listed below, no need to repeat in the description

File: source/_integrations/music_assistant.markdown [hunk 2]
```diff
@@ -135,6 +135,89 @@ Transfer the player's queue to another player.
...
+
+Perform a local search on the Music Assistant library.
+
+- **Data attribute**: `config_entry_id`
+  - **Optional**: No.
+  - **Description**: The Music Assistant instance that the search will be performed upon. Allows for multiple servers to be running.
+  - **Example**: `Music Assistant`
+- **Data attribute**: `media_type`
+  - **Optional**: No.
+  - **Description**: The type of content to search for. Select from artist, album, track, radio, or playlist.
```
Comment (marcelveldt) [inline source/_integrations/music_assistant.markdown:181]:
```suggestion
  - **Description**: The type of content to search for.
```

File: source/_integrations/music_assistant.markdown [hunk 3]
```diff
@@ -135,6 +135,89 @@ Transfer the player's queue to another player.
...
+  - **Optional**: Yes.
+  - **Description**: From what point in the list should results be returned.
+  - **Example**: `10`
+- **Data attribute**: `search`
+  - **Optional**: Yes.
+  - **Description**: A string that will further filter the results.
+  - **Example**: `Home`
+- **Data attribute**: `order_by`
+  - **Optional**: Yes.
+  - **Description**: Sort the list by this field. View available sorting options in the Developer Tools > Actions > music_assistant.get_library action.
```
Comment (marcelveldt) [inline source/_integrations/music_assistant.markdown:201]: The UI will show the available sorting options so not sure why this text is needed ?

File: source/_integrations/music_assistant.markdown [hunk 4]
```diff
@@ -135,6 +135,89 @@ Transfer the player's queue to another player.
...
+  - **Example**: `year`
+- **Data attribute**: `album_artists_only`
+  - **Optional**: Yes.
+  - **Description**: When `artist` is the `media_type` then this option will restrict the result to album artists only.
+  - **Example**: `true`
+- **Data attribute**: `album_type`
+  - **Optional**: Yes.
+  - **Description**: When `album` is the `media_type` then this option will restrict the result according to the selection of either album, single, compilation, EP or unknown.
```
Comment (marcelveldt) [inline source/_integrations/music_assistant.markdown:209]: The UI will display the available options so no need to repeat in the description

**Comment by marcelveldt** [comment]
One thing I'm not sure of is if we should also add the expected output as these actions have return values. @frenck ?

**Comment by frenck** [comment]
I think that would be a nice bonus @marcelveldt, but that can also be a follow up PR (considering beta cut is around the corner).

**Review by marcelveldt — APPROVED** [review]
Nice work @OzGav !

**Comment by marcelveldt** [comment]
@c0ffeeca7 any idea why this change doesnt show up in the current stable docs ?
Maybe it got missed while cherry picking ?

**Comment by c0ffeeca7** [comment]
> @c0ffeeca7 any idea why this change doesnt show up in the current stable docs ? Maybe it got missed while cherry picking ?

<https://github.com/home-assistant/home-assistant.io/pull/36784>

**Pull request merged**